### ISSUE TITLE
Notary services do not resolve "mysql" alias correctly. 

### ISSUE THREAD

**Issue by dbarkelew** [goharbor/harbor#6031]
Harbor notary-server and notary-signer are resolving the "mysql" alias to the resolv.conf search path instead of to "mysql." for the notary-db bridge IP. Looks like a bug caused by libnetwork ignoring ndot:n. 
https://github.com/docker/libnetwork/pull/2212

**Reproduction Steps**
1. Add a "mysql" record to the external DNS. In my case was "mysql.gsslabs.org"
2. Deploy harbor as containers using defaults. I used the VIC Appliance for this.
3. Notary-server logs reports
```
Oct 12 16:28:58 172.18.0.1 notary-server[457]: notaryserver database failed to come up within 30 seconds
Oct 12 16:29:02 172.18.0.1 notary-server[457]: waiting for mysql://server@tcp(mysql:3306)/notaryserver to come up.
```

**Workaround**
Found changing "mysql" to "mysql." for the db_url in the signer-config.json, server-config.json and migrate.sh fixes the issue. Look something like this. 
```
        "storage": {
                "backend": "mysql",
                "db_url": "server@tcp(mysql.:3306)/notaryserver?parseTime=True"
        },
```

Requesting for the  default "mysql" to be changed to "mysql." in the harbor configs.templates for the next release used in the VIC Appliance as a workaround. 
https://github.com/goharbor/harbor/blob/master/make/common/templates/notary/server-config.json
https://github.com/goharbor/harbor/blob/master/make/common/templates/notary/signer-config.json
Only seeing the migrate.sh scripts in the binary package use in the make. 

Bugzilla 2214121

**Comment by reasonerjt** [comment]
But why the it queries the external DNS to resolve mysql given that there is an `alias` in the docker network `notary-mdb`?
https://github.com/goharbor/harbor/blob/release-1.5.0/make/docker-compose.notary.tpl#L54

**Comment by reasonerjt** [comment]
Seems the queries were due to the search domain on `resolv.conf` in the container.
@wy65701436 I think we can set dns_search via docker-compose template to overwrite the value from docker host.
https://docs.docker.com/compose/compose-file/compose-file-v2/#dns_search

**Comment by ywk253100** [comment]
The root cause is a bug in Go DNS resolver before golang 1.8 https://github.com/golang/go/issues/15419, the Notary in Harbor is built with golang 1.7.3. So I think @wy65701436 upgrading the base image should fix this.

**Comment by wy65701436** [comment]
close it as fixed, will provide a new release on v1.5.0 for VIC.
